### ISSUE TITLE
create a simple demo tool for recording gps location data

### ISSUE THREAD

**Issue by spyhunter99** [osmdroid/osmdroid#659]
non-transmission recording of gps data from the demo app that can export to kml or perhaps to gpx or csv

**Comment by monsieurtanuki** [comment]
Good idea! (for the record I'm thinking about creating an open source running app)
Would it be displayed as a new `Activity` in the demo app (part of "More Samples / Location"), or do you have an already existing one in mind?

**Comment by monsieurtanuki** [comment]
I see 3 distinct parts:
1. `Location`s need to be stored in order. In memory the obvious choice is `List<Location>`, but we definitely also need a kind of semi-permanent storage between the begin and the end of the GPS location capture. Something like a new `interface LocationContainer` with methods `List<Location> load()` (to be called in `Fragment.onResume`) and `void save()` (to be called in `Fragment.onPause`). May we use `Parcelable` to store `Location`s - how relevant and safe is it?
1. `List<Location>` need to be exported (as kml, kmz, csv, gpx, ...). Something like a new `interface LocationExporter` with method `void export(List<Location>)`
1. A Fragment where we can begin the GPS location capture, end it, view the path on the map and export the path potentially to different formats

**Comment by spyhunter99** [comment]
yup something like that. the "save your location" actions should be in a background service. Storage in a simple sqlite database would probably be ideal. Just a simple time, lat, lon, speed, heading would do it. Then the activity or sample fragment, would be a simple, start/stop, export/save as option. It could probably be added to the base sample fragment so that it's accessible via the ... menu
